core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -702,16 +713,20 @@ public boolean load(String graphHopperFolder) {
 
         setGraphHopperLocation(graphHopperFolder);
 
-        if (encodingManager == null)
-            setEncodingManager(EncodingManager.create(encodedValueFactory, flagEncoderFactory, ghLocation));
+        if (encodingManager == null) {
+            StorableProperties properties = EncodingManager.loadProperties(ghLocation);
+            encodingManager = properties == null ?
+                    createEncodingManager()
+                    : EncodingManager.create(encodedValueFactory, flagEncoderFactory, properties);
+        }
 
         if (!allowWrites && dataAccessType.isMMap())
             dataAccessType = DAType.MMAP_RO;
 
         GHDirectory dir = new GHDirectory(ghLocation, dataAccessType);
         ghStorage = new GraphHopperStorage(dir, encodingManager, hasElevation(), encodingManager.needsTurnCostsSupport(), defaultSegmentSize);
 
-        checkProfilesConsistency();
+        checkProfilesConsistency(encodingManager);
```

Comment:
Not necessary. But it makes it more obvious that the method has always a non-null EncodingManager (?)

Reply:
Ok I see, it makes it clear that the method depends on encodingManager and this is more critical than e.g. profilesByName, because encodingManager is not final. Hm but it also suggests that the method can/should be used with different encoding managers.. Well, this is probably a matter of taste. The real issue is probably that GraphHopper loads itself and EncodingManager can be null in the first place, but this is out of scope for this PR :)

Reply:
Will revert it. We could avoid this problem if we build the EncodingManager private before it is used and do not expose it via a setter. But the getEncodingManager is still used where it is hard to remove it.

Reply:
Yes sure this is out of scope here. I think in the long run there should be the following *functions*

* config.yml -> GraphHopperConfig
* GraphHopperConfig -> New Import / -gh Folder and maybe an in-memory version of this like GraphHopperConfig -> GraphHopper 
* -gh Folder -> GraphHopper
*  GraphHopperBuilder -> GraphHopper for advanced configurations via Java API that cannot be realized by a config object only.

In any case there should be no class that gets somehow configured, leaves some fields null and then 'loads itself'